### ISSUE TITLE
File paths included in binaries

### ISSUE THREAD

**Issue by lllama** [Nuitka/Nuitka#807]
The compiled binaries include strings with file paths from the build environment.

e.g. On Windows
```
> C:\tools\sysinternals\strings64.exe .\__helpers.obj | rg "\.h\s*$"
C:\tools\sysinternals\strings64.exe .\paths_test.build\__helpers.obj | rg "\.h"
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/dictionaries.h
C:\Users\lllama\DOCUME~1\DEVELO~1\NUITKA~1\VENV~1\lib\SITE-P~1\nuitka\build\include\nuitka/helper/raising.h
```

On Linux the output is different but full paths are included:
```
$ strings paths_test.dist/paths_test  | grep lllam
bytescloselevelrangethrowformatlocals__all____cmp____doc__compilegetattrglobalsinspect__dict____exit____file____iter____main____name____path____spec__fromlist__class____enter__bytearray__cached____import____loader____module____package__classmethod__builtins____compiled__staticmethod__metaclass___initializing__mro_entries____class_getitem__submodule_search_locations/mnt/c/Users/lllama/Documents/development/nuitka-test/.venv_linux/bin/python
```

The following script produces a minimal example:
```
import os
print(os.environ)
```

built with the following command:
Windows
```
 nuitka3.exe --standalone --show-progress --show-scons .\paths_test.py
```
Linux:
```
nuitka3 --standalone --show-progress --show-scons paths_test.py
```

**Comment by kayhayen** [comment]
This is the `sys.executable` path, which is left in there. Probably also is on Windows, if you use a virtualenv.

Having it there for standalone may or may not make all that much sense. Some programs use it to execute Python code, which is working relatively nicely if it happens to be `/usr/bin/python` or something like that.

**Comment by kayhayen** [comment]
So, looking into Windows, I noticed that the file paths of the header files are there, and while checking, I didn't see any usage of `__FILE__` that could explain this, but also duplication of some strings. Then reviewing the MSVC option we use for Nuitka, adding comments, I noticed we use /Ox which does not include duplicate string elimination. And low and behold, it also eliminates unused strings it seems, since the header file references are gone adding ``/GF`` option. Wow.

But there are also references to the file locations of the standard library, i.e. for frozen files, for which we use the `.pyc files it seems`in preference over the `.py` files. These use constants in their marshal, which have absolute paths. Nuitka patches those away when loading them, to be relative to the distribution folder, but that means they are in the binary still. 

That reveals your original Python installation location, which in case of virtualenv, may also be your home directory, compilation path, and contain secrets. Not really acceptable. I will push the first the 2 solutions I mentioned, and then check how to make use of existing bytecode files, while modifying them as well to not include `__file__`, potentially `__path__` and while at it `__doc__` constants, the later for size, to get not only smaller binaries, but no information leaked.

**Comment by kayhayen** [comment]
It seems we were compile all source code with `compile` for at least the Python3 already, sometimes doing source code patches, and we were providing the original filename, instead of a one based on module name only.

That fixes it. As for `__doc__`, that would have to be done in another way then. I will need to check with Python 2.7 separately, if we use source code there, or bytecode.

**Comment by kayhayen** [comment]
Python2 is equally good now, seems that bytecode only files would be treated that way, but that never really happens with standard library.

**Comment by kayhayen** [comment]
This was released as part of 0.6.9 already.
